Sources/Filters/Sources/FrustumSource/index.js

```diff
@@ -0,0 +1,290 @@
+        nbPts += 5;
+      }
+    }
+
+    let pointType = 0;
+    if (model.outputPointsPrecision === DesiredOutputPrecision.SINGLE) {
+      pointType = VtkDataTypes.FLOAT;
+    } else if (model.outputPointsPrecision === DesiredOutputPrecision.DOUBLE) {
+      pointType = VtkDataTypes.DOUBLE;
+    }
+    const newPoints = vtkPoints.newInstance({ dataType: pointType });
+
+    newPoints.setNumberOfPoints(nbPts);
+
+    const pt = [0.0, 0.0, 0.0];
+    const planes = [0, 0, 0];
+
+    planes[0] = 0; // left
+    planes[1] = 2; // bottom
+    planes[2] = 5; // near
+    computePoint(planes, pt);
+    newPoints.setPoint(0, ...pt);
+
+    planes[0] = 1;
+    computePoint(planes, pt);
+    newPoints.setPoint(1, ...pt);
+
+    planes[1] = 3;
+    computePoint(planes, pt);
+    newPoints.setPoint(2, ...pt);
+
+    planes[0] = 0;
+    computePoint(planes, pt);
+    newPoints.setPoint(3, ...pt);
+
+    planes[1] = 2;
+    planes[2] = 4; // far
+    computePoint(planes, pt);
+    newPoints.setPoint(4, ...pt);
+
+    planes[0] = 1;
+    computePoint(planes, pt);
+    newPoints.setPoint(5, ...pt);
+
+    planes[1] = 3;
+    computePoint(planes, pt);
+    newPoints.setPoint(6, ...pt);
+
+    planes[0] = 0;
+    computePoint(planes, pt);
+    newPoints.setPoint(7, ...pt);
+
+    const numPolys = 6;
+    const newPolys = vtkCellArray.newInstance();
+    newPolys.allocate(numPolys * 5);
+
+    let pts = [];
+
+    // left
+    pts = [4, 0, 3, 7];
+    newPolys.insertNextCell(pts);
+
+    // right
+    pts = [1, 5, 6, 2];
+    newPolys.insertNextCell(pts);
+
+    // bottom
+    pts = [0, 4, 5, 1];
+    newPolys.insertNextCell(pts);
+
+    // top
+    pts = [3, 2, 6, 7];
+    newPolys.insertNextCell(pts);
+
+    // near
+    pts = [0, 1, 2, 3];
+    newPolys.insertNextCell(pts);
+
+    // far
+    pts = [4, 7, 6, 5];
+    newPolys.insertNextCell(pts);
+
+    let newLines = null;
+    if (model.showLines) {
+      const numLines = 4;
+      newLines = vtkCellArray.newInstance();
+      newLines.allocate(numLines * 3);
+
+      pts = [12, 8]; // apex, or first of the two extra near points.
+
+      // line from lower-left corner
+      if (parallelFrustum) {
+        pts[0] = 0;
+      }
+      pts[1] = 8;
+      newLines.insertNextCell(pts);
+
+      // line from lower-right corner
+      if (parallelFrustum) {
+        pts[0]++;
+      } else if (leftRightNull) {
+        pts[0] = 13;
+      }
+      pts[1]++;
+      newLines.insertNextCell(pts);
+
+      // line from upper-right corner
+      if (parallelFrustum) {
+        pts[0]++;
+      } else if (bottomTopNull) {
+        pts[0] = 13;
+      }
+      pts[1]++;
+      newLines.insertNextCell(pts);
+
+      // line from upper-left corner
+      if (parallelFrustum) {
+        pts[0]++;
+      } else if (leftRightNull) {
+        pts[0] = 12;
+      }
+      pts[1]++;
+      newLines.insertNextCell(pts);
+    }
+
+    output.setPoints(newPoints);
+
+    if (newLines) {
+      output.setLines(newLines);
```

Comment:
you could move `if (newLines) output.setLines()` in the `if(showLines)` block above.